public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +62,289 @@ const TITLE_TEXT_DICT = {
+          <EuiButton iconType="popout" iconSide="right">
+            Create new permission group
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </FormRow>
+  );
+}
+
+function DocLevelSecurityRow(props: {
+  value: string;
+  onChangeHandler: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Document level security"
+      headerSubText="You can restrict a role to a subset of document in an index."
+      helpLink="/"
+      optional
+    >
+      <EuiTextArea
+        placeholder={FIELD_LEVEL_SECURITY_PLACEHOLDER}
+        value={props.value}
+        onChange={props.onChangeHandler}
+      />
+    </FormRow>
+  );
+}
+
+function FieldLevelSecurityRow(props: {
+  method: FieldLevelSecurityMethod;
+  fields: OptionSeletion;
+  onMethodChangeHandler: (s: string) => void;
+  onFieldChangeHandler: (s: OptionSeletion) => void;
+  onFieldCreateHandler: (s: string) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Field level security"
+      headerSubText="You can restrict what document field that user can see. If you use field level security in conjunction with document-level security, make sure you don't restrict to the field that document-level security uses."
+      optional
+    >
+      <EuiFlexGroup>
+        <EuiFlexItem grow={1}>
+          <EuiSuperSelect
+            valueOfSelected={props.method}
+            options={[
+              { inputDisplay: 'Include', value: 'include' },
+              { inputDisplay: 'Exclude', value: 'exclude' },
+            ]}
+            onChange={props.onMethodChangeHandler}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={9}>
+          <EuiComboBox
+            noSuggestions
+            placeholder="Type in field name"
+            selectedOptions={props.fields}
+            onChange={props.onFieldChangeHandler}
+            onCreateOption={props.onFieldCreateHandler}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </FormRow>
+  );
+}
+
+function AnonymizationRow(props: {
+  value: OptionSeletion;
+  onChangeHandler: (s: OptionSeletion) => void;
+  onCreateHandler: (s: string) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Anonymization"
+      headerSubText="Masks any sensitive fields with random value to protect your data security."
+      optional
+    >
+      <EuiComboBox
+        noSuggestions
+        placeholder="Type in field name"
+        selectedOptions={props.value}
+        onChange={props.onChangeHandler}
+        onCreateOption={props.onCreateHandler}
+      />
+    </FormRow>
+  );
+}
+
+function generateIndexPermissionPanels(
+  indexPermissions: RoleIndexPermissionStateClass[],
+  permisionOptionsSet: EuiComboBoxOptionOption[],
+  setRoleIndexPermission: Dispatch<SetStateAction<RoleIndexPermissionStateClass[]>>
+) {
+  const panels = indexPermissions.map((perm, index) => {
+    const onValueChangeHandler = (field: string) =>
```

Comment:
Originally I thought this is related to field-level security. But this is actually the rowName for every row you render?